Xamarin.Forms.Core/Visuals/VisualTypeConverter.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Xamarin.Forms
+{
+	[Xaml.TypeConversion(typeof(IVisual))]
+	public class VisualTypeConverter : TypeConverter
+	{
+		static Dictionary<string, IVisual> _visualTypeMappings;
+		static Dictionary<string, IVisual> InitMappings()
+		{
+			var mappings = new Dictionary<string, IVisual>(StringComparer.OrdinalIgnoreCase);
+			_visualTypeMappings = mappings;
+			return mappings;
+		}
+
+		public static void Register(string visualKey, IVisual visual)
+		{
+			if (_visualTypeMappings == null)
+				InitMappings();
+
+			_visualTypeMappings[visualKey] = visual;
+		}
+
+		internal static void Register(Type visual)
+		{
+			if (_visualTypeMappings == null)
+				InitMappings();
+
+			if(!_visualTypeMappings.ContainsKey(visual.FullName))
+			{
+				try
+				{
+					IVisual registeredVisual = (IVisual)Activator.CreateInstance(visual);
```

Comment:
Not crazy about this. Need to see if there's a better way to do this. Just not sure yet how to pull out the IVisual type for the converter to use